nano/core_test/bootstrap.cpp

```diff
@@ -310,15 +310,13 @@ TEST (bootstrap_processor, process_one)
 
 	node_config.peering_port = nano::test::get_available_port ();
 	node_flags.disable_rep_crawler = true;
-	auto node1 (std::make_shared<nano::node> (system.io_ctx, nano::unique_path (), node_config, system.work, node_flags));
+	auto node1 = system.add_node (node_config, node_flags);
```

Comment:
I think the reason it is constructing a nano::node instead of calling ad_node, is so that the test can control the establishment of tcp connections. Although, I am not convinced that the test is correct but at least that seems to be the motivation behind it. to create a disconnected node that it then chooses to connect by calling bootstrap_INITIATOR.BOOTSTRAP()